examples-gallery/plot_two_link_pendulum_on_a_cart.py

```diff
@@ -0,0 +1,275 @@
+Upright a double pendulum.
+==========================
+
+A double pendulum is rotably attached to a cart which can move
+along the horizontal X axis. The goal is to get the double
+pendulum to an upright position in the shortest time possible,
+given an upper limit on the absolute value of the force that
+can be applied to the cart.
+Gravity points in the negative Y direction.
+
+**Constants**
+
+- m1:     mass of the cart [kg]
+- m2:     mass of the first pendulum [kg]
+- m3:     mass of the second pendulum [kg]
+- lx:     length of the first pendulum rods [m]
+- iZZ1:   moment of inertia of the first pendulum about its end point [kg*m^2]
+- iZZ2:   moment of inertia of the second pendulum about its end point [kg*m^2]
+- g:      acceleration due to gravity [m/s^2]
+
+**States**
+
+- q1:     position of the cart along the X axis [m]
+- q2:     angle of the first pendulum with respect to the vertical [rad]
+- q3:     angle of the second pendulum with respect to the first [rad]
+- u1:     speed of the cart along the X axis [m/s]
+- u2:     angular speed of the first pendulum [rad/s]
+- u3:     angular speed of the second pendulum [rad/s]
+
+**Specifieds**
+
+- f:      force applied to the cart [N]
+
+"""
+import sympy.physics.mechanics as me
+from collections import OrderedDict
+import numpy as np
+import sympy as sm
+from opty.direct_collocation import Problem
+
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+from matplotlib import patches
+
+# %%
+# Generate the nonholonomic equations of motion of the system.
+N, A1, A2 = sm.symbols('N A1, A2', cls = me.ReferenceFrame)
+t = me.dynamicsymbols._t
+O, P1, P2, P3  = sm.symbols('O P1 P2 P3', cls = me.Point)                               
+O.set_vel(N, 0)                                                                        
+q1, q2, q3, u1, u2, u3, F = me.dynamicsymbols('q1 q2 q3 u1 u2 u3 F')                   
+lx, m1, m2, m3, g, iZZ1, iZZ2 = sm.symbols('lx, m1, m2, m3 g, iZZ1, iZZ2')             
+
+A1.orient_axis(N, q2, N.z)
+A1.set_ang_vel(N, u2 * N.z)
+A2.orient_axis(N, q3, N.z)
+A2.set_ang_vel(N, u3 * N.z)
+
+P1.set_pos(O, q1 * N.x)
+P2.set_pos(P1, lx * A1.x)
+P2.v2pt_theory(P1, N, A1)
+
+P3.set_pos(P2, lx * A2.x)
+P3.v2pt_theory(P2, N, A2)
+
+P1a = me.Particle('P1a', P1, m1)                                                
+
+I = me.inertia(A1, 0, 0, iZZ1)
+P2a = me.RigidBody('P2a', P2, A1, m2, (I, P2))                                 
+
+I = me.inertia(A2, 0, 0, iZZ2)
+P3a = me.RigidBody('P3a', P3, A2, m3, (I, P3))                                  
+BODY = [P1a, P2a, P3a]
+
+FL = [(P1, F * N.x - m1*g*N.y), (P2, -m2*g*N.y), (P3, -m3*g*N.y)]                   
+kd = sm.Matrix([q1.diff(t) - u1, q2.diff(t) - u2, q3.diff(t) - u3])                   
+
+q_ind = [q1, q2, q3]
+u_ind = [u1, u2, u3]
+
+KM = me.KanesMethod(N, q_ind=q_ind, u_ind=u_ind, kd_eqs=kd)
+(fr, frstar) = KM.kanes_equations(BODY, FL)
+EOM = kd.col_join(fr + frstar)
+sm.pprint(EOM)                
```

Comment:
```suggestion
sm.pprint(sm.trigsimp(EOM))                
```

I think that might make it display more compact equations without too much computational time cost.